### PR TITLE
refactor(github): decouple trellogithub and "github.com/google/go-git…

### LINKED ISSUE #307: `proposal`: decouple `trellogithub` and  "github.com/google/go-github/v42/github"
## Description

Letting the `trellogithub` without any line of code directly use the "github.com/google/go-github/v42/github"

See #277 for more info.

/assign @warren830

### PR DESCRIPTION
**Pull request by warren830**
# Summary

move Repositories.getContents to pkg/util/github to decouple  trellogithub and "github.com/google/go-github/v42/github"

### Related Issues

closes #307

### CODE REVIEW

File: internal/pkg/plugin/trellogithub/github.go [hunk 1]
```diff
@@ -5,12 +5,10 @@ import (
 	"net/http"
 	"strings"
 
-	githubx "github.com/google/go-github/v42/github"
-	"github.com/spf13/viper"
-
 	"github.com/merico-dev/stream/pkg/util/github"
 	"github.com/merico-dev/stream/pkg/util/log"
 	"github.com/merico-dev/stream/pkg/util/mapz"
+	"github.com/spf13/viper"
```
Comment (daniel-hutao) [inline internal/pkg/plugin/trellogithub/github.go:11]:
```go
 	"github.com/spf13/viper"

	"github.com/merico-dev/stream/pkg/util/github"
 	"github.com/merico-dev/stream/pkg/util/log"
 	"github.com/merico-dev/stream/pkg/util/mapz"
```

File: pkg/util/github/github.go [hunk 2]
```diff
@@ -5,7 +5,7 @@ import (
 	"fmt"
 	"os"
 
-	"github.com/google/go-github/v42/github"
+	github "github.com/google/go-github/v42/github"
```
Comment (daniel-hutao) [inline pkg/util/github/github.go:8]: "github.com/google/go-github/v42/github"

File: internal/pkg/plugin/trellogithub/github.go [hunk 3]
```diff
@@ -37,12 +35,11 @@ func (tg *TrelloGithub) VerifyWorkflows(workflows []*github.Workflow) (map[strin
 
 func (tg *TrelloGithub) FetchRemoteContent(wsFiles []string) ([]string, map[string]error, error) {
```
Comment (daniel-hutao) [inline internal/pkg/plugin/trellogithub/github.go:36]: It's better to sink this function into `util/github` package.

File: internal/pkg/plugin/trellogithub/github.go [hunk 4]
```diff
@@ -119,12 +116,11 @@ func (tg *TrelloGithub) AddTrelloIdSecret(trelloId *TrelloItemId) error {
 }
 
 func (tg *TrelloGithub) GetWorkflowPath() (string, error) {
```
Comment (daniel-hutao) [inline internal/pkg/plugin/trellogithub/github.go:118]: It's better to sink this function into `util/github` package.

**Comment by daniel-hutao** [comment]
@warren830 Thanks for your pr, you did a good job with decouple.

However, the operation of github in the `trellogithub` package is still too complicated. This kind of complex logic should be encapsulated in `util/github` package. Otherwise, `util/github` will have no advantage over the `native github` package.

**Comment by daniel-hutao** [comment]
@warren830 I saw your update; thank you for giving up your time off over the weekend to continue contributing. But there are still some problems, such as:

- internal/pkg/plugin/trellogithub/github.go

```go
	"github.com/merico-dev/stream/pkg/util/github"
 	"github.com/spf13/viper"
```

should be

```go
 	"github.com/spf13/viper"

	"github.com/merico-dev/stream/pkg/util/github"
```

- internal/pkg/plugin/trellogithub/helper.go

```go
	"github.com/merico-dev/stream/pkg/util/github"
 	"github.com/merico-dev/stream/pkg/util/trello"
 	"github.com/mitchellh/mapstructure"
```

should be 

```go
 	"github.com/mitchellh/mapstructure"

	"github.com/merico-dev/stream/pkg/util/github"
 	"github.com/merico-dev/stream/pkg/util/trello"
```

- pkg/util/github/github.go

```go
	"github.com/merico-dev/stream/pkg/util/log"
 	"github.com/merico-dev/stream/pkg/util/mapz"
 	"golang.org/x/oauth2"
```

should be 

```go
 	"golang.org/x/oauth2"

	"github.com/merico-dev/stream/pkg/util/log"
 	"github.com/merico-dev/stream/pkg/util/mapz"
```

See the fmt stage in Makefile(`make fmt`) for more info.

File: pkg/util/github/github.go [hunk 5]
```diff
@@ -99,3 +102,59 @@ func NewClient(option *Option) (*Client, error) {
 
 	return client, nil
 }
+
+func FetchRemoteContent(ctx context.Context, owner string, repo string, path string, wsFiles []string) ([]string, map[string]error, error) {
+	var filesInRemoteDir = make([]string, 0)
+	_, dirContent, resp, err := client.Repositories.GetContents(
```
Comment (daniel-hutao) [inline pkg/util/github/github.go:108]: The client is `nil` here. You shouldn't use the `client` directly here, but put this code into a client method, something like: 

```go
func (c *Client) FetchRemoteContent(){c.XXX()}
```

**Review by daniel-hutao — CHANGES_REQUESTED** [review]
See the comment above.

Please don't rush and complete the refactoring slowly. Please test it after writing the code to make sure it works. After the test is passed, welcome to continue call me anytime. Thanks again for your hard work.

**Comment by IronCore864** [comment]
> @warren830 I saw your new commit; Thank you for taking the time to submit contributions over the weekend.
> 
> But there are still some problems, such as:
> 
>     * internal/pkg/plugin/trellogithub/github.go
> 
> 
> ```go
> 	"github.com/merico-dev/stream/pkg/util/github"
>  	"github.com/spf13/viper"
> ```
> 
> should be
> 
> ```go
>  	"github.com/spf13/viper"
> 
> 	"github.com/merico-dev/stream/pkg/util/github"
> ```
> 
>     * internal/pkg/plugin/trellogithub/helper.go
> 
> 
> ```go
> 	"github.com/merico-dev/stream/pkg/util/github"
>  	"github.com/merico-dev/stream/pkg/util/trello"
>  	"github.com/mitchellh/mapstructure"
> ```
> 
> should be
> 
> ```go
>  	"github.com/mitchellh/mapstructure"
> 
> 	"github.com/merico-dev/stream/pkg/util/github"
>  	"github.com/merico-dev/stream/pkg/util/trello"
> ```
> 
>     * pkg/util/github/github.go
> 
> 
> ```go
> 	"github.com/merico-dev/stream/pkg/util/log"
>  	"github.com/merico-dev/stream/pkg/util/mapz"
>  	"golang.org/x/oauth2"
> ```
> 
> should be
> 
> ```go
>  	"golang.org/x/oauth2"
> 
> 	"github.com/merico-dev/stream/pkg/util/log"
>  	"github.com/merico-dev/stream/pkg/util/mapz"
> ```
> 
> See the fmt stage in Makefile(`make fmt`) for more info.

This is the standard against which all future imports order will be measured.

A big thumbs-up to Daniel!

**Review by IronCore864 — APPROVED** [review]

**Review by daniel-hutao — APPROVED** [review]

**Pull request merged**